OBSDOCS-552:  sample service config for Cluster Observability Operator tech preview (#67608)
openshift/openshift-docs: modules/monitoring-deploying-a-sample-service-for-cluster-observability-operator.adoc

// Module included in the following assemblies:
//
// monitoring/cluster-observability-operator/configuring-the-cluster-observability-operator-to-monitor-a-service.adoc

:_mod-docs-content-type: PROCEDURE
[id="deploying-a-sample-service-for-cluster-observability-operator_{context}"]
= Deploying a sample service for Cluster Observability Operator

> This sample configuration deploys a sample service named `prometheus-coo-example-app` in the user-defined `ns1-coo` project. 

Author: Changed

The service exposes the custom `version` metric.

.Prerequisites

* You have access to the cluster as a user with the `cluster-admin` cluster role or as a user with administrative permissions for the namespace.

.Procedure

> . Create a YAML file named `prometheus-coo-example-app.yaml` that contains the following configuration details for a namespace, deployment, and service::

Author: Fixed

// Module included in the following assemblies:
//
// monitoring/cluster-observability-operator/configuring-the-cluster-observability-operator-to-monitor-a-service.adoc

:_mod-docs-content-type: PROCEDURE
[id="deploying-a-sample-service-for-cluster-observability-operator_{context}"]
= Deploying a sample service for Cluster Observability Operator

This sample configuration deploys a sample service named `prometheus-coo-example-app` in the user-defined `ns1-coo` project. 
The service exposes the custom `version` metric.

.Prerequisites

* You have access to the cluster as a user with the `cluster-admin` cluster role or as a user with administrative permissions for the namespace.

.Procedure

> . Create a service configuration YAML file named `prometheus-coo-example-app.yaml`.
> 
> . Add the following deployment and service configuration details to the file:

Reviewer: ```suggestion
. Create a YAML file named `prometheus-coo-example-app.yaml` that contains the following configuration details for a namespace, deployment, and service:
```

suggestion for minimalism

Author: Excellent suggestion. Done.

+
[source,yaml]
----
apiVersion: v1
kind: Namespace
metadata:
  name: ns1-coo
---
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app: prometheus-coo-example-app
  name: prometheus-coo-example-app
  namespace: ns1-coo
spec:
  replicas: 1
  selector:
    matchLabels:
      app: prometheus-coo-example-app
  template:
    metadata:
      labels:
        app: prometheus-coo-example-app
    spec:
      containers:
      - image: ghcr.io/rhobs/prometheus-example-app:0.4.1
        imagePullPolicy: IfNotPresent
        name: prometheus-coo-example-app
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: prometheus-coo-example-app
  name: prometheus-coo-example-app
  namespace: ns1-coo
spec:
  ports:
  - port: 8080
    protocol: TCP
    targetPort: 8080
    name: web
  selector:
    app: prometheus-coo-example-app
  type: ClusterIP
----

. Save the file.

> . Apply the configuration from the file to the cluster:
> +
> [source,terminal]
> ----
> $ oc apply -f prometheus-coo-example-app.yaml
> ----

Reviewer: ```suggestion
. Apply the configuration to the cluster by running the following command:
+
[source,terminal]
----
$ oc apply -f prometheus-coo-example-app.yaml
----
```

Author: Changed

> . Verify that the pod is running:

Reviewer: ```suggestion
. Verify that the pod is running by running the following command and observing the output:
```

Author: Added

// Module included in the following assemblies:
//
// monitoring/cluster-observability-operator/configuring-the-cluster-observability-operator-to-monitor-a-service.adoc

:_mod-docs-content-type: PROCEDURE
[id="deploying-a-sample-service-for-cluster-observability-operator_{context}"]
= Deploying a sample service for Cluster Observability Operator

This sample configuration deploys a sample service named `prometheus-coo-example-app` in the user-defined `ns1-coo` project. 
The service exposes the custom `version` metric.

.Prerequisites

> * You have access to the cluster as a user with the `cluster-admin` cluster role or the `monitoring-edit` cluster role.

Reviewer 2: A user with cluster role `monitoring-edit` has no enough permission to create APP service and can only manage the following resources
```
 - servicemonitors
  - podmonitors
  - prometheusrules

Reviewer 2: admin role of specific namespace can create app service in it

Reviewer 3: +1 I think we might want to avoid referencing roles that are created by CMO. That just creates unnecessary coupling.
A namespace admin role would look like this
```
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
 name: read-write
 namespace: ns1-coo
rules:
- apiGroups:
 - ""
 resources: ["*"]
 verbs:
 - get
 - list
 - watch
 - create
 - update
 - patch
 - delete
```
Though maybe there is a convention is the docs already?

Author: @jan--f @lihongyan1 -

What do you think of this wording:

`* You have access to the cluster as a user with the `cluster-admin` cluster role or as a user with administrative permissions for the namespace.`

Reviewer 2: The changes look good.

> * You have installed the Cluster Observability Operator.

Reviewer 2: Deploying a sample service doesn't need COO has been installed, this line can be deleted

Author: Removed.

.Procedure

. Create a service configuration YAML file named `prometheus-coo-example-app.yaml`.

. Add the following deployment and service configuration details to the file:
+
[source,yaml]
----
apiVersion: v1
kind: Namespace
metadata:
  name: ns1-coo
---
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app: prometheus-coo-example-app
  name: prometheus-coo-example-app
  namespace: ns1-coo
spec:
  replicas: 1
  selector:
    matchLabels:
      app: prometheus-coo-example-app
  template:
    metadata:
      labels:
        app: prometheus-coo-example-app
    spec:
      containers:

>       - image: ghcr.io/rhobs/prometheus-coo-example-app:0.4.1

Reviewer 3: ```suggestion
      - image: ghcr.io/rhobs/prometheus-example-app:0.4.1
```
The image link doesn't change compared to the `Managing Metrics` section in the existing docs.

Author: Fixed
